webapp/src/Service/StatisticsService.php

```diff
@@ -514,6 +517,103 @@ public function getGroupedProblemsStats(
+     *          teams: array<int, Team>,
+     *          team_count: int,
+     *          solved: int,
+     *          not_solved: int,
+     *          total: int,
+     *          problems_solved: array<int, ContestProblem>,
+     *          problems_solved_count: int,
+     *          problems_attempted: array<int, ContestProblem>,
+     *          problems_attempted_count: int,
+     *     }>
+     * }
+     */
+    public function getLanguagesStats(Contest $contest, string $view): array
+    {
+        /** @var Language[] $languages */
+        $languages = $this->em->getRepository(Language::class)
+            ->createQueryBuilder('l')
+            ->andWhere('l.allowSubmit = 1')
+            ->orderBy('l.name')
+            ->getQuery()
+            ->getResult();
+
+        $languageStats = [];
+
+        foreach ($languages as $language) {
+            $languageStats[$language->getLangid()] = [
+                'name' => $language->getName(),
+                'teams' => [],
+                'team_count' => 0,
+                'solved' => 0,
+                'not_solved' => 0,
+                'total' => 0,
+                'problems_solved' => [],
+                'problems_solved_count' => 0,
+                'problems_attempted' => [],
+                'problems_attempted_count' => 0,
+            ];
+        }
+
+        $teams = $this->getTeams($contest, $view);
+        foreach ($teams as $team) {
+            foreach ($team->getSubmissions() as $s) {
+                if ($s->getContest() != $contest) {
+                    continue;
+                }
+                if ($s->getSubmitTime() > $contest->getEndTime()) {
+                    continue;
+                }
+                if ($s->getSubmitTime() < $contest->getStartTime()) {
+                    continue;
+                }
+                if ($s->getSubmittime() > $contest->getFreezetime()) {
+                    continue;
+                }
+
+                $language = $s->getLanguage();
+
+                $languageStats[$language->getLangid()]['teams'][$team->getTeamid()] = $team;
+                $languageStats[$language->getLangid()]['total']++;
+                if ($s->getResult() === 'correct') {
+                    $languageStats[$language->getLangid()]['solved']++;
+                    $languageStats[$language->getLangid()]['problems_solved'][$s->getProblem()->getProbId()] = $s->getContestProblem();
+                } else {
+                    $languageStats[$language->getLangid()]['not_solved']++;
+                }
+                $languageStats[$language->getLangid()]['problems_attempted'][$s->getProblem()->getProbId()] = $s->getContestProblem();
+            }
+        }
+
+        foreach ($languageStats as &$languageStat) {
+            usort($languageStat['teams'], static fn(
+                Team $a,
+                Team $b
+            ) => ($a->getLabel() ?: $a->getExternalid()) <=> ($b->getLabel() ?: $b->getExternalid()));
```

Comment:
Sort by number of problems solved using this language instead?

Reply:
Do you think that makes more sense than by team ID? Won't it look confusing if the teams are in arbitrary order?

Or do you propose to also show the number of solved problems per team per language? What if a team didn't solve but did try a problem in the language?